src/Tools/Source/RunTests/Cache/ContentUtil.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace RunTests.Cache
+{
+    internal sealed class ContentUtil
+    {
+        private readonly Options _options;
+        private readonly MD5 _hash = MD5.Create();
+        private readonly Dictionary<string, string> _fileToChecksumMap = new Dictionary<string, string>();
```

Comment:
None of this stuff is locked.  Is there any concern about them being called concurrently?